libs/books/feature/src/lib/book-search/book-search.component.spec.ts

```diff
@@ -26,55 +26,50 @@ describe('BookSearchComponent', () => {
         SharedTestingModule
       ],
       providers: [
-        Store,
         provideMockStore({
           initialState: {
             books: {
               entities: []
             }
-          },
-          selectors: [{
-            selector: getAllBooks,
-            value: []
-          }]
+          }
         }),
       ],
     }).compileComponents();
   }));
+
   beforeEach(() => {
     fixture = TestBed.createComponent(BookSearchComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
-    store = fixture.debugElement.injector.get(Store);
+    store = TestBed.inject(MockStore);
   });
 
   describe('ngOnInit()', () => {
     beforeEach(() => {
       spyOn(component.searchForm, 'valueChanges');
       spyOn(store, 'dispatch');
-      spyOn(store, 'select').and.returnValue(BOOKS);
     });
 
-    it('It should call searchBooks() when we have value for search term and actual time spent after entering search term is equal to 500ms', fakeAsync(() => {
+    it('searchBooks() should be called when we have empty value for search term and actual time spent after entering search term is equal to 500ms', fakeAsync(() => {
       component.ngOnInit();
       component.searchForm.controls.term.setValue('Angular');
+
       tick(500);
+
       component.searchForm.valueChanges.subscribe(() => {
-        expect(store.dispatch).toHaveBeenCalledTimes(1);
         expect(store.dispatch).toHaveBeenCalledWith(searchBooks({ term: 'Angular' }))
       });
     }));
 
-    it('It should populate the array of Book object', fakeAsync(() => {
+    it('clearSearch() should be called when we have value for search term and actual time spent after entering search term is equal to 500ms', fakeAsync(() => {
```

Comment:
Please change it to 
`No action should be called when we have value for search term and actual time spent after entering search term is less than 500ms`

Reply:
Done